### PR TITLE
Add option to disable the platform locale

### PR DESCRIPTION
**Pull request by jhnns**
As discussed in https://github.com/sindresorhus/camelcase-keys/issues/81

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -76,29 +76,36 @@ const camelCase = (input, options) => {
...
 	}
 
 	input = input.replace(LEADING_SEPARATORS, '');
 
 	if (options.preserveConsecutiveUppercase) {
-		input = preserveConsecutiveUppercase(input);
+		input = preserveConsecutiveUppercase(input, toLowerCase);
 	} else {
-		input = input.toLocaleLowerCase();
+		input = toLowerCase(input);
```
Comment (jhnns) [inline index.js:101]: I think this fixes a bug "by accident" where the locale hasn't been applied previously.

File: test.js [hunk 2]
```diff
@@ -204,8 +204,43 @@ test('camelCase with locale option', t => {
...
+	});
+
+	try {
+		fn();
+	} finally {
+		Object.defineProperty(String.prototype, 'toLocaleUpperCase', toLocaleUpperCase);
+		Object.defineProperty(String.prototype, 'toLocaleLowerCase', toLocaleLowerCase);
+	}
+};
+/* eslint-enable no-extend-native */
```
Comment (jhnns) [inline test.js:246]: I tried to separate this mocking setup from the rest of the test. I'm not sure if you agree with this style of mocking String.prototype functions. Another solution would be to spin up a separate Node process with `tr-TR` as locale but I thought that would be a bigger change.

File: readme.md [hunk 3]
```diff
@@ -60,8 +60,16 @@ camelCase(['__foo__', '--bar'], {pascalCase: true});
...
+camelCase('lorem-ipsum') // on a platform with 'en-US'
 camelCase('lorem-ipsum', {locale: 'en-US'});
 //=> 'loremIpsum'
+
+camelCase('lorem-ipsum') // on a platform with 'tr-TR'
+camelCase('lorem-ipsum', {locale: 'tr-TR'});
+//=> 'loremİpsum'
+
+camelCase('lorem-ipsum', {locale: false}); // ignore platform locale
+//=> 'loremIpsum'
```
Comment (sindresorhus) [inline readme.md:72]: I don't think these additions are needed. It's enough to document it at the API docs for the option.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
You also need to update index.d.ts

**Comment by jhnns** [comment]
I've updated the PR, ready to be reviewed again :)

**Pull request merged**